SillyBusinessInc/Moldbreaker
# Feat: #181 make the ranged enemy

## Purpose of this PR:
making a range enemy based on pim's code, the enemy should attack x amount times then go to a different location then start attacking again or doing another state or something based on where the player is.

## How to test:
go to spawner scene in the map mustafa check all ranged enemy code.  check if he shoots x amount of times then does something different goes to different location and just check if the enemy is what is expected of an enemy.

### links: 
[Ticket](https://github.com/SillyBusinessInc/Moldbreaker/issues/181)
